src/table/body-cell/index.tsx

```diff
@@ -56,6 +56,28 @@ function TableCellEditable<ItemType>({
   const [hasFocus, setHasFocus] = useState(false);
   const showIcon = hasHover || hasFocus;
 
+  const prevSuccessfulEdit = useRef(successfulEdit);
+  const prevHasFocus = useRef(hasFocus);
+  const [showSuccessIcon, setShowSuccessIcon] = useState(false);
+
+  useEffect(() => {
+    // Early return if there hasn't been any successfulEdit state changes.
+    if (!successfulEdit && !prevSuccessfulEdit.current) {
+      return;
+    }
+
+    // Show the success icon right after a successful edit (when successfulEdit switches to true)
```

Comment:
The comments are describing the opposite of what's happening inside the `if` blocks